scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,410 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Module to generate a column map to their corresponding statistical variable defintions.
+"""
+import os
+import io
+import sys
+import logging
+import json
+import re
+import csv
+from zipfile import ZipFile
+from collections import OrderedDict
+
+# Allows the following module imports to work when running as a script
+_SCRIPT_PATH = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(
+    _SCRIPT_PATH, '../../../../../util/'))  # for statvar_dcid_generator
+
+from statvar_dcid_generator import get_stat_var_dcid
+
+# intitalize the logger
+logging.basicConfig(
+    level=logging.ERROR,
+    filename=os.path.join(_SCRIPT_PATH, 'make_column_map.log'),
+    format="{asctime} {processName:<12} {message} ({filename}:{lineno})",
+    style="{",
+    filemode='w')
+logging.StreamHandler(stream=None)
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.CRITICAL)
+
+# list of properties that every statistical variable should contain
+_MANDATORY_PROPS = ['measuredProperty', 'populationType', 'statType']
+
+# list of keys expected in the JSON spec, if they are not defined, we set as an
+# empty dictionary
+_JSON_KEYS = [
+    'populationType', 'measurement', 'pvs', 'inferredSpec',
+    'enumSpecializations', 'pvs', 'universePVs', 'ignoreColumns',
```

Comment:
'pvs' repeated

Reply:
Thanks for pointing this out. I have removed it